docs/internal/pnpm-migration-plan.md

```
@@ -0,0 +1,939 @@
+
+- [ ] **Step 2: Run a full install to materialize and reconcile the lockfile**
+
+```bash
+pnpm install
+```
+Expected: completes and writes/updates `pnpm-lock.yaml`. **Do not** pass `--frozen-lockfile` here (we are intentionally generating it). Note any `WARN` lines about unmet peer deps or ignored build scripts — see Step 3 and Task 11.
+
+- [ ] **Step 3: Allowlist dependency build scripts (`strictDepBuilds` is on by default)**
+
+pnpm 11 does not run dependency lifecycle scripts (`postinstall`, etc.) unless allowlisted, and `strictDepBuilds: true` (the default) makes an **unreviewed** build script a **hard install failure** — `pnpm install` exits 1 with `ERR_PNPM_IGNORED_BUILDS`. The `onlyBuiltDependencies` setting from pnpm 9/10 was **removed**; the replacement is the `allowBuilds:` map in `pnpm-workspace.yaml`. Inspect what is blocked:
+```bash
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should be surrounded by blank lines

(MD031, blanks-around-fences)